src/env_cli_vars.nim

```diff
@@ -0,0 +1,146 @@
+import os, sequtils, strutils, strformat, chronicles
+
+import # vendor libs
+  confutils
+
+const envVarsPrefix = "STATUS"
+const buildTimePrefix = "BUILD_"
+const runTimePrefix = "RUNTIME_"
+
+proc defaultDataDir*(): string =
+  try:
+    let homeDir = getHomeDir()
+    let parentDir =
+      if defined(development):
+        parentDir(getAppDir())
+      elif homeDir == "":
+        getCurrentDir()
+      elif defined(macosx):
+        joinPath(homeDir, "Library", "Application Support")
+      elif defined(windows):
+        let targetDir = getEnv("LOCALAPPDATA").string
+        if targetDir == "":
+          joinPath(homeDir, "AppData", "Local")
+        else:
+          targetDir
+      else:
+        let targetDir = getEnv("XDG_CONFIG_HOME").string
+        if targetDir == "":
+          joinPath(homeDir, ".config")
+        else:
+          targetDir
+    return absolutePath(joinPath(parentDir, "Status"))
+  except OSError:
+    echo "Error: Unable to determine home directory."
+
+type StatusDesktopConfig = object
+  # build vars
+  infuraToken* {.
+    defaultValue: ""
+    desc: "Sets infura token"
+    name: buildTimePrefix & "INFURA_TOKEN"
+    abbr: "infura-token" .}: string
+  infuraTokenSecret* {.
+    defaultValue: ""
+    desc: "Sets infura token secret"
+    name: buildTimePrefix & "INFURA_TOKEN_SECRET"
+    abbr: "infura-token-secret" .}: string
+  poktToken* {.
+    defaultValue: ""
+    desc: "Sets pokt token"
+    name: buildTimePrefix & "POKT_TOKEN"
+    abbr: "pokt-token" .}: string
+  openSeaApiKey* {.
+    defaultValue: ""
+    desc: "Sets open sea api key"
+    name: buildTimePrefix & "OPENSEA_API_KEY"
+    abbr: "open-sea-api-key" .}: string
+  alchemyEthereumMainnetToken* {.
+    defaultValue: ""
+    desc: "Sets alchemy ethereum mainnet token"
+    name: buildTimePrefix & "ALCHEMY_ETHEREUM_MAINNET_TOKEN"
+    abbr: "alchemy-ethereum-mainnet-token" .}: string
+  alchemyEthereumGoerliToken* {.
+    defaultValue: ""
+    desc: "Sets alchemy ethereum goerli token"
+    name: buildTimePrefix & "ALCHEMY_ETHEREUM_GOERLI_TOKEN"
+    abbr: "alchemy-ethereum-goerli-token" .}: string
+  alchemyArbitrumMainnetToken* {.
+    defaultValue: ""
+    desc: "Sets alchemy arbitrum mainnet token"
+    name: buildTimePrefix & "ALCHEMY_ARBITRUM_MAINNET_TOKEN"
+    abbr: "alchemy-arbitrum-mainnet-token" .}: string
+  alchemyArbitrumGoerliToken* {.
+    defaultValue: ""
+    desc: "Sets alchemy arbitrum goerli token"
+    name: buildTimePrefix & "ALCHEMY_ARBITRUM_GOERLI_TOKEN"
+    abbr: "alchemy-arbitrum-goerli-token" .}: string
```

Comment:
This looks to me like you are applying the `STATUS_BUILD_` prefix to these CLI flags which is still wrong.

I will attempt to explain it for the last and final time.

* `STATUS_BUILD_` flags should only have effect at build time.
* `STATUS_RUNTIME_` flags should only have effect at runtime.
* CLI flags should NOT include anything build time related.
* Either remove build-time only flags from CLI flags, or make it so that `STATUS_BUILD_` variables work at build time and `STATUS_RUNTIME_` works at runtime.

Lets take an example:

* `STATUS_BUILD_ALCHEMY_ARBITRUM_GOERLI_TOKEN` should work __only at build time__.
* `STATUS_RUNTIME_ALCHEMY_ARBITRUM_GOERLI_TOKEN` should work __only at runtime__.